docs/implementation.md

```diff
@@ -61,9 +63,157 @@ records such that
 
 In this storage, index updation is a no-op, since the bloom filters are already written as a part of committing data.
 
+In the case of Copy-On-Write, a single parquet file constitutes one `file slice` which contains one complete version of 
+the file 
+
+{% include image.html file="hoodie_log_format_v2.png" alt="hoodie_log_format_v2.png" max-width="1000" %}
+
 #### Merge On Read
 
-Work in Progress .. .. .. .. ..
+In the Merge-On-Read storage model, the processes(spark application) for ingesting data (both inserts/updates) into the dataset
+ and that of creating compacted views are decoupled. The former is hereby referred to as `Writer` while the later
+ is referred as `Compactor`
+ 
+##### Merge On Read Writer
+ 
+ At a high level, Merge-On-Read Writer goes through same stages as Copy-On-Write writer in ingesting data.
+ The key difference here is that updates are appended to latest log (delta) file belonging to the latest file slice 
+ without merging. For inserts, Hudi supports 2 modes:
+
+   1. Inserts To Log Files - For external indices which are capable of looking up record-keys that could have 
```

Comment:
May be say : 

1. Inserts to Log Files - This is done for datasets that have an indexable log files (for eg global index)
2. Inserts to parquet files - This is done for datasets that do not have indexable log files, for eg bloom index embedded in parquer files...